api/health/handler.go

```diff
@@ -45,14 +47,32 @@ func NewGetAndPostHandler(log logging.Logger, reporter Reporter) (http.Handler,
 	return handler, err
 }
 
+type errorMsg struct {
+	Error string `json:"error"`
+}
+
 // NewGetHandler return a health handler that supports GET requests reporting
 // the result of the provided [reporter].
-func NewGetHandler(reporter func() (map[string]Result, bool)) http.Handler {
+func NewGetHandler(reporter func(tags ...string) (map[string]Result, bool)) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// Make sure the content type is set before writing the header.
 		w.Header().Set("Content-Type", "application/json")
 
-		checks, healthy := reporter()
+		subnetIDs := r.URL.Query()["subnetID"]
```

Comment:
Kinda weird that the key is `subnetID` and it returns `subnetIDs`... But I do think this is the best way to do this with the query params...

Reply:
yea and it's even weirder that `r.URL.Query().Get("subnetID")` returns the first value for the key... alternatively we can use plural `subnetIDs` (now `tags`) and split string value by comas. but I probably would prefer singular key names (i.e `tag` over `tags`).